### DISCUSSION TITLE
DDT4all ECU read/write

### DISCUSSION THREAD

**Discussion by dtiberiu** [cedricp/ddt4all#938] [Ideas]
Hi everyone,
First of all, allow me to introduce myself; I´m an electronic engineer pasionate of cars and I have a project consisting in replacing an Delphi ECU by a Siemens SID 301 ECU on an old Dacia Logan and do an immo off on the SID301 ECU.
I tried to read the EEPROM of the SID 301 usind DDT4all but I cannot find this function. I was able to comunicate with the ECU on bench, I read the VIN and other information from ECU but I have not found the way to Read/Write ECU.
SID ECU Hardware
Microcontroller – FREESCALE MPC561
External FLASH memory – AM29BDD160GB (Memory size – 2 Megabytes)
External EEPROM memory – 95320 (Memory size – 4 Kilobytes)

Looking forward for your help.
Thank you.

### Discussed in https://github.com/cedricp/ddt4all/discussions/922

<div type='discussions-op-text'>

<sup>Originally posted by **alitarasi** October 15, 2023</sup>
Hi everyone
 I install the latest version of DDT4all. but after installation it did not work. what should I do? </div>

**Comment by Marianpol** [comment]
There is no such functionality in DDT4ALL. This tool is not intended to do that kind of stuff.  

I'm gonna give you some ideas how you can achive your goals:
- try ReadMemoryByAddress/WriteMemoryByAddress request to read/write EEPROM and FLASH (maybe Security Access is not required to at least dump these data)
- sniff CAN data when reading/writing FLASH using e.g KESSv2 to get an idea what requests are used to read/wirte FLASH
- disassemble SID301 BDM FLASH to get an idea what is the Security Access algorithm to be able to read/write FLASH
- disassemble Renault Immo Tool - it has all functionalities that you need
